wordpress-mobile/WordPress-iOS
# Blaze: Customize web view nav bar based on the current step

Part of #20095
Part of #20096
Ref: pe7hp4-b3-p2

## Description
This PR customizes the nav bar button based on the current step. The button is disabled during non-dismissible states, and the text is changed to "Done" when the ad creation is complete.

## Notes

The following areas are not handled yet and will be addressed in a future PR:
* Analytics tracking
* Reachability handling
* Blocking external URLs

### Recording

https://user-images.githubusercontent.com/25306722/221956358-21b38134-ec63-424e-af11-577a33b843fe.mp4

## Testing Instructions

1. Open the app
2. Navigate to the Menu
3. Tap on "Blaze"
4. Make sure the blaze site web flow is presented in a modal
5. Verify that the navigation bar button is enabled and reads "Cancel"
6. Choose a post to promote
7. Verify that the navigation bar button is enabled and reads "Cancel"
8. Fill in the fields in the "Appearance" step and tap "Next" (Indicate in the title that this is for testing to avoid it being approved)
9. Verify that the navigation bar button is enabled and reads "Cancel"
10. Tap "Next"
11. Verify that the navigation bar button is enabled and reads "Cancel"
12. Fill in payment details or choose an existing payment, then tap "Next"
13. The web view should display "Checking payment information" for a while if you've added a new payment
14. P.S: There's a known issue where the dismiss button is not disabled during the "Checking payment information" step. (Ref: p1677558200851969-slack-C04LJFS1G5P)
15. Next, the web view should display "Creating your ad" for a while
16. Verify that the navigation bar button is disabled and reads "Cancel" during this step
17. Next, the web view should display "All set!"
18. Verify that the navigation bar button is enabled and reads "Done"
19. Tap "Done"
20. Verify that the modal is dismissed

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
22 unit tests under `BlazeWebViewModelTests`

PR submission checklist:
